src/App.tsx

```diff
@@ -1,51 +1,127 @@
 import React from 'react';
+import classNames from 'classnames';
+import { v4 as uuidv4 } from 'uuid';
 import './App.scss';
 
-// import goodsFromServer from './goods';
-
-export const App: React.FC = () => {
-  return (
-    <main className="App">
-      <header className="App__header">
-        <h1 className="App__title">
-          Jam is selected
-        </h1>
-
-        <button
-          type="button"
-          className="App__clear"
-        >
-          Clear
-        </button>
-      </header>
-
-      <ul>
-        <li className="Good">Dumplings</li>
-        <li className="Good">Carrot</li>
-        <li className="Good">Eggs</li>
-        <li className="Good">Ice cream</li>
-        <li className="Good">Apple</li>
-        <li className="Good">Bread</li>
-        <li className="Good">Fish</li>
-        <li className="Good">Honey</li>
-        <li className="Good Good--active">Jam</li>
-        <li className="Good">Garlic</li>
-      </ul>
-
-      {/* Put required buttons into each Good */}
-      <button
-        type="button"
-        className="Good__remove"
-      >
-        Remove
-      </button>
-
-      <button
-        type="button"
-        className="Good__select"
-      >
-        Select
-      </button>
-    </main>
-  );
+import goodsFromServer from './goods';
+
+const uniqueKeys: string[] = [];
+
+for (let i = 0; i < goodsFromServer.length; i += 1) {
+  uniqueKeys[i] = uuidv4();
+}
+
+type State = {
+  selectedGoods: string[];
+  title: string;
 };
+
+export class App extends React.Component<{}, State> {
+  state: Readonly<State> = {
+    selectedGoods: ['Jam'],
+    title: '',
+  };
+
+  componentDidMount() {
+    this.setTitle();
+  }
+
+  setTitle = () => {
+    const { selectedGoods } = this.state;
+    const { length } = selectedGoods;
+
+    if (length === 0) {
+      this.setState({ title: 'No goods selected' });
+    }
+
+    if (length === 1) {
+      this.setState({ title: `${selectedGoods[0]} is selected` });
+    }
+
+    if (length > 1) {
+      this.setState({ title: `${selectedGoods.slice(0, length - 1).join(', ')} and ${selectedGoods[length - 1]} are selected` });
+    }
+  };
+
+  handlingRemove = (good: string, goods: string[]) => {
+    this.setState(
+      { selectedGoods: goods.filter(item => item !== good) },
+      this.setTitle,
```

Comment:
All handlers call the setTitle method, maybe it's better to call it inside render, it will work in all cases

Reply:
not agree